server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeHealth.java

```diff
@@ -0,0 +1,305 @@
+                Map<String, String> nodeAttributes = nodeDiscovery.getAttributes();
+                if (!nodeAttributes.isEmpty()) {
+                    if (nodeAttributes.containsKey(awarenessAttributeName)) {
+                        attributeValue = nodeAttributes.get(awarenessAttributeName);
+
+                        if (!perAttributeValueHealthMap.containsKey(attributeValue)) {
+                            clusterAwarenessAttributeValueHealth = new ClusterAwarenessAttributeValueHealth(attributeValue);
+                            clusterAwarenessAttributeValueHealth.setNodes(1);
+                            clusterAwarenessAttributeValueHealth.setActiveShards(shardAllocationPerNode.getOrDefault(node, 0));
+                            double attributeWeight = 1.0;
+                            if (weightedRoutingWeights != null) {
+                                attributeWeight = weightedRoutingWeights.getOrDefault(attributeValue, 1.0);
+                            }
+                            clusterAwarenessAttributeValueHealth.setWeight(attributeWeight);
+                        } else {
+                            clusterAwarenessAttributeValueHealth = perAttributeValueHealthMap.get(attributeValue);
+                            clusterAwarenessAttributeValueHealth.setNodes(clusterAwarenessAttributeValueHealth.getNodes() + 1);
+                            clusterAwarenessAttributeValueHealth.setActiveShards(
+                                clusterAwarenessAttributeValueHealth.getActiveShards() + shardAllocationPerNode.getOrDefault(node, 0)
+                            );
+                        }
+                        perAttributeValueHealthMap.put(attributeValue, clusterAwarenessAttributeValueHealth);
+                    }
+                }
+            }
+        }
+
+        generatePerAttributeValueHealthList(perAttributeValueHealthMap, totalShards, replicaEnforcement, forcedZoneSettingsExists);
+    }
+
+    private void generatePerAttributeValueHealthList(
+        Map<String, ClusterAwarenessAttributeValueHealth> perAttributeValueHealthMap,
+        int totalShards,
+        boolean replicaEnforcement,
+        boolean forcedZoneSettingsExists
+    ) {
+
+        int numberOfAttributeValue = perAttributeValueHealthMap.size();
+        int totalShardsDistributedPerAttributeValue = 0;
+
+        // Can happen customer has defined weights as well as awareness attribute but no node level attribute was there
+        // So to avoid divide-by-zero error checking this
+        if (numberOfAttributeValue != 0) {
+            totalShardsDistributedPerAttributeValue = totalShards / numberOfAttributeValue;
+        }
+
+        for (String attributeValueKey : perAttributeValueHealthMap.keySet()) {
+            ClusterAwarenessAttributeValueHealth attributeHealthMapPerValue = perAttributeValueHealthMap.get(attributeValueKey);
+
+            if (replicaEnforcement && forcedZoneSettingsExists) {
```

Comment:
what happens if discovered zone and forced zone awareness is mutually exclusive?

Reply:
Replica balance is only guaranteed if we set both the properties otherwise that would not take affect